### ISSUE TITLE
Sometimes the transport_tcp listener doesn't see a discovered_nat message

### ISSUE THREAD

**Issue by Vagabond** [helium/erlang-libp2p#357]
When a NAT is discovered, the nat server sends the transport listener a nat_discovered message. Sometimes, however, the message is not received. I do not know why this is.

**Comment by wolasss** [comment]
@Vagabond Could this be a reason why p2p connection fails with timeout?

**Comment by Vagabond** [comment]
No, that's likely due to a routing failure, the target being too busy or a stale peerbook.
